src/opencl_bitcracker_fmt_plug.c

```diff
@@ -0,0 +1,630 @@
+	HANDLE_CLERROR(ciErr1, "clEnqueueWriteBuffer"); 
+
+	ciErr1 = clEnqueueWriteBuffer(queue[gpu_id], devicePasswordSize, CL_TRUE, 0, MAX_KEYS_PER_CRYPT * sizeof(int), inbuffer_size, 0, NULL, NULL);      
+	HANDLE_CLERROR(ciErr1, "clEnqueueWriteBuffer");
+
+    hostFound[0] = -1;
+    ciErr1 = clEnqueueWriteBuffer(queue[gpu_id], deviceFound, CL_TRUE, 0, sizeof(int), hostFound, 0, NULL, NULL);      
+	HANDLE_CLERROR(ciErr1, "clEnqueueWriteBuffer"); 
+
+    ciErr1 = clSetKernelArg(crypt_kernel, 0, sizeof(cl_int), (void*)&numPassword);
+	HANDLE_CLERROR(ciErr1, "clSetKernelArg"); 
+
+    ciErr1 |= clSetKernelArg(crypt_kernel, 1, sizeof(cl_mem), (void*)&devicePassword);
+	HANDLE_CLERROR(ciErr1, "clSetKernelArg"); 
+
+	ciErr1 |= clSetKernelArg(crypt_kernel, 2, sizeof(cl_mem), (void*)&devicePasswordSize);
+	HANDLE_CLERROR(ciErr1, "clSetKernelArg"); 
+
+    ciErr1 |= clSetKernelArg(crypt_kernel, 3, sizeof(cl_mem), (void*)&deviceFound);
+	HANDLE_CLERROR(ciErr1, "clSetKernelArg"); 
+
+    ciErr1 |= clSetKernelArg(crypt_kernel, 4, sizeof(cl_mem), (void*)&deviceEncryptedVMK);
+	HANDLE_CLERROR(ciErr1, "clSetKernelArg"); 
+
+    ciErr1 |= clSetKernelArg(crypt_kernel, 5, sizeof(cl_mem), (void*)&w_blocks_d);
+	HANDLE_CLERROR(ciErr1, "clSetKernelArg"); 
+
+    ciErr1 |= clSetKernelArg(crypt_kernel, 6, sizeof(cl_int), (void*)&IV0);
+	HANDLE_CLERROR(ciErr1, "clSetKernelArg"); 
+
+    ciErr1 |= clSetKernelArg(crypt_kernel, 7, sizeof(cl_int), (void*)&IV4);
+	HANDLE_CLERROR(ciErr1, "clSetKernelArg"); 
+
+    ciErr1 |= clSetKernelArg(crypt_kernel, 8, sizeof(cl_int), (void*)&IV8);
+	HANDLE_CLERROR(ciErr1, "clSetKernelArg"); 
+
+    ciErr1 |= clSetKernelArg(crypt_kernel, 9, sizeof(cl_int), (void*)&IV12);
+	HANDLE_CLERROR(ciErr1, "clSetKernelArg"); 
+
+    time (&start);
+    ciErr1 = clEnqueueNDRangeKernel(queue[gpu_id], crypt_kernel, 1, NULL, &szGlobalWorkSize, &szLocalWorkSize, 0, NULL, NULL);
+	HANDLE_CLERROR(ciErr1, "clEnqueueNDRangeKernel"); 
+
+    ciErr1 = clEnqueueReadBuffer(queue[gpu_id], deviceFound, CL_TRUE, 0, sizeof(unsigned int), hostFound, 0, NULL, NULL);
+	HANDLE_CLERROR(ciErr1, "clEnqueueNDRangeKernel"); 
+    HANDLE_CLERROR(clFlush(queue[gpu_id]), "clFlush"); 
+    time (&end);
+    dif = difftime (end,start);
+
+    printf ("[BitCracker] -> Attack stats: %d passwords evaluated in %.2lf seconds => %.2f pwd/s\n",  numPassword, dif, (double)(numPassword/dif) );
+
+	HANDLE_CLERROR(clFinish(queue[gpu_id]), "clFinish"); 
+
+    totPsw += numPassword;
+    
+	HANDLE_CLERROR(clReleaseKernel(crypt_kernel), "[BitCracker] ->  Release prepare_kernel");
+
+    if(w_blocks_d)clReleaseMemObject(w_blocks_d);
+    if(devicePassword)clReleaseMemObject(devicePassword);
+	if(devicePasswordSize)clReleaseMemObject(devicePasswordSize);
+    if(deviceEncryptedVMK)clReleaseMemObject(deviceEncryptedVMK);
+    if(deviceFound)clReleaseMemObject(deviceFound);
+
+	return count;
+}
```

Comment:
You shouldn't create/init kernel and buffers in crypt_all(). It should be created in `init()`, auto-tuned in `reset()` and destroyed in `done()`.